File: sdks/php/src/Resources/Alerts.php
<?php

declare(strict_types=1);

namespace HookSniff\Resources;

use HookSniff\Request;

/**
 * HookSniff API Resource: Alerts
 */
class Alerts
{
    /**
     * @param array{baseUrl: string, token: string, timeout: int, numRetries: int} $ctx
     */
    public function __construct(private readonly array $ctx) {}

    /**
     * List alert rules.
     *
     * @return array<int, array<string, mixed>>
     * @throws \HookSniff\ApiException
     */
    public function listRules(): array
    {
        $req = new Request('GET', '/v1/alerts/rules');
        return $req->send($this->ctx);
    }

    /**
     * List alert notifications.
     *
     * @return array<int, array<string, mixed>>
     * @throws \HookSniff\ApiException
     */
    public function listNotifications(?int $limit = null): array
    {
        $req = new Request('GET', '/v1/alerts/notifications');
        if ($limit !== null) {
            $req->setQueryParams(['limit' => $limit]);
        }
        return $req->send($this->ctx);
    }
}

File: sdks/rust/src/resources/alerts.rs
//! HookSniff API Resource: Alerts
//!
//! Alert rules and notifications.

use crate::request::{HookSniffRequest, HookSniffRequestContext, HttpMethod};
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct AlertRule {
    pub id: String,
    pub name: String,
    pub condition: String,
    pub threshold: f64,
    pub enabled: bool,
    pub created_at: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct AlertNotification {
    pub id: String,
    pub rule_id: String,
    pub message: String,
    pub read: bool,
    pub created_at: String,
}

#[derive(Debug)]
pub struct Alerts {
    ctx: HookSniffRequestContext,
}

impl Alerts {
    pub fn new(ctx: HookSniffRequestContext) -> Self {
        Self { ctx }
    }

    /// Get alert rules
    pub fn rules(&self) -> Result<Vec<AlertRule>, Box<dyn std::error::Error>> {
        let req = HookSniffRequest::new(HttpMethod::Get, "/v1/alerts/rules");
        req.send(&self.ctx)
    }

    /// Get alert notifications
    pub fn notifications(&self) -> Result<Vec<AlertNotification>, Box<dyn std::error::Error>> {
        let req = HookSniffRequest::new(HttpMethod::Get, "/v1/alerts/notifications");
        req.send(&self.ctx)
    }
}
